### ISSUE TITLE
(dev2.0) More currying issues

### ISSUE THREAD

**Issue by Alxandr** [fable-compiler/Fable#1459]
### Description

Not sure how to classify this one (previous one was easier to identify, given that it happened with base constructors), but the following code produces wrong function calls:

### Repro code

```fsharp
type BaseClass (f: string -> string -> string) =
  member __.MakeString a b = f a b

type AddString (f: string -> string -> string) =
  inherit BaseClass (fun a b -> f a b + f b a)

let s = (AddString (fun a b -> "Prefix: " + a + b)).MakeString "a" "b"
```

```js
export function BaseClass(f) {
  const $this$$1 = this;
  $this$$1.f = f;
}
export function BaseClass$$$$002Ector(f) {
  return this != null ? BaseClass.call(this, f) : new BaseClass(f);
}
export function BaseClass$$MakeString(__, a, b) {
  return __.f(a, b);
}
export function AddString(f$$1) {
  BaseClass$$$$002Ector.call(this, function (a$$1, b$$1) {
    // AddString is called with `(string, string) => string`, but
    // here it's invoked as if it's curried.
    return f$$1(a$$1)(b$$1) + f$$1(b$$1)(a$$1);
  });
}
inherits(AddString, BaseClass);
export function AddString$$$$002Ector(f$$1) {
  return this != null ? AddString.call(this, f$$1) : new AddString(f$$1);
}

// Function here is not curried
export const s = BaseClass$$MakeString(AddString$$$$002Ector(function (a$$2, b$$2) {
  return "Prefix: " + a$$2 + b$$2;
}), "a", "b");
```

As can be seen above (I've inserted comments in the JS), there is some wrong currying going on here.

### Related information

* Fable version (`dotnet fable --version`): 2.0.0-alpha-026
* Operating system: Windows 10

**Comment by alfonsogarciacaro** [comment]
Thanks for catching this! I shouldn't have moved the base call out of the constructor body 😅 It should be fixed in 2.0.0-alpha-027 :+1:

**Comment by Alxandr** [comment]
Great. My tests are passing again, except for 2 but those are probably my fault. I'll try removing some of the hacks/workarounds I did for fable 1 to see if things have been fixed in Fable 2 (like the fact that you couldn't have internal constructors of singleton cases) and let you know what breaks 😛.
